cow_instrument/MoveCommand.cpp

```diff
@@ -3,8 +3,10 @@
 
 MoveCommand::MoveCommand(V3D offset) : m_offset(offset) {}
 
-void MoveCommand::execute(Component &component) const {
-  component.deltaPos(m_offset);
+bool MoveCommand::execute(CowPtr<Component> &component) const {
+  component->deltaPos(m_offset); // De-referenceing forces a copy
+  return true; // If you're moving, components are being written to. Copy
+               // happened on previous line.
```

Comment:
@SimonHeybrock. Yes my comment is misleading, and yes, I should do something like `auto unique_component = component.unique()` prior to the write and return `!unique_component`.